rust/processor/src/config.rs

```diff
@@ -27,6 +27,8 @@ pub struct IndexerGrpcProcessorConfig {
     pub auth_token: String,
     // Version to start indexing from
     pub starting_version: Option<u64>,
+    // Version to start indexing from if nothing in db
+    pub starting_version_if_nothing_in_db: Option<u64>,
```

Comment:
we probably want a better name..

also, for these two parameters, it's not clear which one will take precedence. we probably should either use an enum here, or ensure they are mutually exclusive.

Reply:
I believe we should avoid using enums to maintain backward compatibility. The current implementation works as follows:
- If starting_version exists in the config, it will override all existing values (both starting_version_if_nothing_in_db and the database value).
- If starting_version is not configured, we'll take a look to the database value. If the database has no value and starting_version_if_nothing_in_db is configured, we'll use that value; otherwise, we'll default to 0.